### PR TITLE
Backport of #3540 Repeat to kill the process of a canceled task.

### LINKED ISSUE #3407: Task cancellation gets stuck
**Version**
Pulpcore 3.16 via Satellite 6.11

**Describe the bug**
If "Reclaim Space" (Actions::Pulp3::CapsuleContent::ReclaimSpace) action is executed and canceled after that task will get stuck in canceling not only in foreman tasks but also in pulp3. 

The task was apparently stuck in "canceling" state for more than 6 days.

**To Reproduce**
How reproducible:
Always

Steps to Reproduce:
1. Execute "Reclaim Space" on Satellite
2. Cancel task (Actions::Pulp3::CapsuleContent::ReclaimSpace)

**Expected results:**
Task will get canceled

**Actual results:**
Task is stuck in foreman-tasks and in pulp3

**Additional context**
https://bugzilla.redhat.com/show_bug.cgi?id=2143290

### PR DESCRIPTION
**Pull request by decko**
fixes #3407

(cherry picked from commit dea600fe6674e68dd7c6fdd50630f000af8773bb)

### CODE REVIEW

File: pulpcore/tests/functional/api/test_tasking.py [hunk 1]
```diff
@@ -1,277 +0,0 @@
-"""Tests related to the tasking system."""
```
Comment (mdellweg) [inline pulpcore/tests/functional/api/test_tasking.py:1]: It doesn't look wise to delete the tests for the very thing you are changing.

Comment (ggainey) [reply pulpcore/tests/functional/api/test_tasking.py:1]: @mdellweg IIRC (apologies for letting these PRs languish), the problem is that the associated tests rely on fixtures that haven't been backported, and we (i.e., I) didn't want to open a can of "just one more thing" cherrypicks just to add the tests - esp for a small, well-contained fix.

Comment (mdellweg) [reply pulpcore/tests/functional/api/test_tasking.py:1]: What am I missing here? I see this PR deleting the file `test_tasking.py` from the 3.22 branch. Are you saying that backporting that fix is invalidating _all_ the tests we have on this branch for the tasking system?

Comment (mdellweg) [reply pulpcore/tests/functional/api/test_tasking.py:1]: This file exists on the 3.22 branch. And in the current state this PR attempts to delete it. And with it these tests:
```
test_multi_resource_locking
test_delete_cancel_waiting_task
test_delete_cancel_running_task
test_cancel_delete_finished_task
test_cancel_nonexistent_task
test_retrieve_task_with_limited_fields
test_retrieve_task_without_specific_fields
test_retrieve_task_with_minimal_fields
test_retrieve_task_using_invalid_worker
test_retrieve_task_using_valid_worker
test_retrieve_task_using_invalid_date
test_retrieve_task_using_valid_date
test_search_task_by_name
test_search_task_using_an_invalid_name
test_filter_tasks_using_worker__in_filter
```
They can't _all_ be invalidated by this fix.

Comment (decko) [reply pulpcore/tests/functional/api/test_tasking.py]: my bad @mdellweg. Something went wrong with the cherry-pick and we didn't noticed until it's late.
Hope it's alright now.

File: pulpcore/tasking/pulpcore_worker.py [hunk 2]
```diff
@@ -162,8 +167,8 @@ def worker_cleanup(self):
 
     def beat(self):
         if self.worker.last_heartbeat < timezone.now() - timedelta(seconds=self.heartbeat_period):
-            self.worker = handle_worker_heartbeat(self.name)
-            if self.shutdown_requested:
+            self.worker = self.handle_worker_heartbeat()
```
Comment (mdellweg) [inline pulpcore/tasking/pulpcore_worker.py:170]: This line should not have changed.
```suggestion
            self.worker = handle_worker_heartbeat(self.name)
```

**Review by mdellweg — APPROVED** [review]

**Pull request merged**